be/src/formats/parquet/page_reader.cpp

```diff
@@ -31,25 +29,24 @@ Status PageReader::next_header() {
 
     uint32_t header_length = 0;
     size_t nbytes = kHeaderBufSize;
-
+    size_t remaining = _finish_offset - _offset;
     do {
-        RETURN_IF_ERROR(_stream.get_bytes(&page_buf, &nbytes, true));
+        nbytes = std::min(nbytes, remaining);
+        RETURN_IF_ERROR(_stream->get_bytes(&page_buf, _offset, &nbytes));
```

Comment:
without the peek, the data read size might double, you should merge @GavinMar bug fix